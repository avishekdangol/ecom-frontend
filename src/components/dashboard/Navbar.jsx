import { Col, Row } from 'antd';
import { TiHome } from 'react-icons/ti';
import { Link } from 'react-router-dom';
import { getUserData } from '@/utils/common';
import UserMenu from '@/components/navbar/components/UserMenu';

function Navbar() {
  const me = getUserData?.value;
  return (
    <Row className="justify-between items-center">
      <Col>
        <Link
          to="/"
          className="flex items-center"
        >
          <TiHome className="text-2xl mb-1" />
          {' '}
          <h3 className="ml-1">Pahichaan</h3>
        </Link>
      </Col>

      <Col>
        <div className="flex items-center">
          <div className="text-right">
            <h3>{me?.name}</h3>
            <p className="capitalize mb-0">{me?.role}</p>
          </div>
          <UserMenu />
        </div>
      </Col>
    </Row>
  );
}

export default Navbar;
