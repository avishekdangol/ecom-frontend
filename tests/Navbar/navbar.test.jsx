import { screen } from '@testing-library/react';
import { render } from '@root/tests/testUtils';
import Navbar from '@/components/navbar/Index.jsx';

describe('Navbar', () => {
  test('snapshot when user isn\'t logged in', () => {
    const { asFragment } = render(<Navbar isLoggedIn={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('snapshot when user is logged in', () => {
    const { asFragment } = render(<Navbar isLoggedIn />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('if user is not logged in, they should see login | register button', () => {
    render(<Navbar isLoggedIn={false} />);
    const component = screen.getByText(/login | register/i);
    expect(component).toBeInTheDocument();
  });

  test('if user is logged in, they should see usermenu button', () => {
    render(<Navbar isLoggedIn />);
    const component = screen.queryByText(/login | register/i);
    expect(component).not.toBeInTheDocument();
  });
});
