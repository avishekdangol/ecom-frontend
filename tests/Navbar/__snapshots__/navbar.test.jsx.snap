// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="ant-row justify-between css-dev-only-do-not-override-1g853jt"
  >
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="ant-row items-center css-dev-only-do-not-override-1g853jt"
      >
        <div
          class="ant-col mr-3 css-dev-only-do-not-override-1g853jt"
        >
          <section
            class="dropdown"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-round ant-btn-default flex items-center justify-center"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="me-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 7a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 13a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Menu
              </span>
            </button>
          </section>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1g853jt"
        >
          <span
            class="ant-input-affix-wrapper search-bar rounded-2xl css-dev-only-do-not-override-1g853jt"
          >
            <span
              class="ant-input-prefix"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <line
                  x1="21"
                  x2="16.65"
                  y1="21"
                  y2="16.65"
                />
              </svg>
            </span>
            <input
              class="ant-input css-dev-only-do-not-override-1g853jt"
              placeholder="Search"
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="flex justify-center items-center"
      >
        <a
          href="/"
        >
          <img
            alt=""
            class="logo cursor-pointer"
            src="/assets/images/logo.png"
            width="94"
          />
        </a>
      </div>
    </div>
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="flex justify-end items-center h-full"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-link px-2"
          type="button"
        >
          <svg
            class="text-black hover:text-slate-500"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-link px-2"
          type="button"
        >
          <svg
            class="text-black hover:text-slate-500"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </button>
        <div
          class="ant-space css-dev-only-do-not-override-1g853jt ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small ant-dropdown-trigger"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-avatar ant-avatar-circle cursor-pointer bg-teal-700 flex items-center ml-4 css-dev-only-do-not-override-1g853jt"
            >
              <span
                class="ant-avatar-string"
                style="transform: scale(1) translateX(-50%);"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.5 6.5C7.5 8.981 9.519 11 12 11s4.5-2.019 4.5-4.5S14.481 2 12 2 7.5 4.019 7.5 6.5zM20 21h1v-1c0-3.859-3.141-7-7-7h-4c-3.86 0-7 3.141-7 7v1h17z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar snapshot when user isn't logged in 1`] = `
<DocumentFragment>
  <div
    class="ant-row justify-between css-dev-only-do-not-override-1g853jt"
  >
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="ant-row items-center css-dev-only-do-not-override-1g853jt"
      >
        <div
          class="ant-col mr-3 css-dev-only-do-not-override-1g853jt"
        >
          <section
            class="dropdown"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-round ant-btn-default flex items-center justify-center"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="me-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 7a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 13a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
              <span>
                Menu
              </span>
            </button>
          </section>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-1g853jt"
        >
          <span
            class="ant-input-affix-wrapper search-bar rounded-2xl css-dev-only-do-not-override-1g853jt"
          >
            <span
              class="ant-input-prefix"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <line
                  x1="21"
                  x2="16.65"
                  y1="21"
                  y2="16.65"
                />
              </svg>
            </span>
            <input
              class="ant-input css-dev-only-do-not-override-1g853jt"
              placeholder="Search"
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="flex justify-center items-center"
      >
        <a
          href="/"
        >
          <img
            alt=""
            class="logo cursor-pointer"
            src="/assets/images/logo.png"
            width="94"
          />
        </a>
      </div>
    </div>
    <div
      class="ant-col ant-col-8 css-dev-only-do-not-override-1g853jt"
    >
      <div
        class="flex justify-end items-center h-full"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-link px-2"
          type="button"
        >
          <svg
            class="text-black hover:text-slate-500"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-link px-2"
          type="button"
        >
          <svg
            class="text-black hover:text-slate-500"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </button>
        <a
          class="ms-4 flex items-center justify-center"
          href="/login"
          shape="round"
        >
          <svg
            class="me-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
            />
          </svg>
          Login | Register
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
